media/src/project/project-map.tsx

```diff
@@ -85,15 +93,30 @@ export const ProjectMap = () => {
             let layers = await Promise.all(
                 layersRsps.map((response: any) => { return response.json(); })
             );
-            setLayerData(layers);
+
+            // Create an empty layer if none exist, otherwise
+            // unpack the event data
+            if (layers.length === 0) {
+                addLayer();
+                setLayerTitleCount((prev) => {return prev + 1;});
+            } else {
+                setLayerData(layers);
+                setActiveLayer(layers[0].pk);
+
+                let events = layers.reduce((acc, val) => {
+                    acc.set(val.pk, val.event_set);
+                    return acc;
+                }, new Map());
+                updateEventData(events);
+            }
         };
 
         getData();
     }, []);
 
-    const addLayer = (title: string) => {
+    const addLayer = () => {
         authedFetch('/api/layer/', 'POST', JSON.stringify(
-            {title: title, content_object: `/api/project/${projectPk}/`}))
+            {title: `Layer ${layerTitleCount}`, content_object: `/api/project/${projectPk}/`}))
```

Comment:
Tracking the number after "Layer" is going to be a bit tricky. If the page is refreshed, the counter resets to 1. The only other way around this chicken/egg issue is to wait until the layer has a valid name before creating it. I think this is more something to think about as you go forward.